tests/testthat/test_add_robust_constraints.R

```diff
@@ -1,19 +1,206 @@
+    add_constant_robust_constraints(x, conf_level) |>
+    prioritizr::add_binary_decisions() %>%    prioritizr::add_default_solver(verbose = F)
+
+  sp2 <- solve(p2)
+
+  feature_rep2 <- eval_feature_representation_summary(p2, sp2)
+  feature_target_a2 <- feature_rep$relative_held[x == 'a'] > target
+  expect_gte(mean(feature_target_a2), conf_level)
+  feature_target_b2 <- feature_rep$relative_held[x == 'b'] > target
+  expect_gte(mean(feature_target_b2), conf_level)
 })
 
 test_that("compile (multiple zones)", {
-  # TODO
+  skip_if_not_installed("terra")
+  # import data
+  sim_pu_raster <- prioritizr::get_sim_pu_raster()
+  sim_features <- prioritizr::get_sim_features()
+  sim_features_z1 <- prioritizr::get_sim_features()
+  sim_features_z2 <- prioritizr::get_sim_features()
+  # define feature groupings
+  x <- rep_len(c("a", "b"), terra::nlyr(sim_features))
+
+  t2 <- matrix(NA, ncol = 2, nrow = 5)
+  t2[, 1] <- 0.1
+  t2[, 2] <- 0.05
+
+  zones <- prioritizr::zones(z1 = sim_features_z1,
+                             z2 = sim_features_z2)
+
+  conf_level <- .6
+  target <- 0.1
+  # build problem
+  p <-
+    prioritizr::problem(c(sim_pu_raster,sim_pu_raster), zones) |>
+    add_robust_min_set_objective(method = "Chance") |>
+    prioritizr::add_relative_targets(t2) |>
+    add_constant_robust_constraints(x, conf_level) |>
+    prioritizr::add_binary_decisions() %>%    prioritizr::add_default_solver(verbose = F)
+
+  cp <- compile(p)
+
+  expect_equal(nrow(cp$A()), length(cp$row_ids()))
+
+  expect_equal(nrow(cp$A()), length(cp$rhs()))
+
+  expect_equal(ncol(cp$A()), length(cp$obj()))
 })
 
 test_that("solve (multiple zones)", {
-  # TODO
+  skip_if_not_installed("terra")
+  # import data
+  sim_pu_raster <- prioritizr::get_sim_pu_raster()
+  sim_features <- prioritizr::get_sim_features()
+  sim_features_z1 <- prioritizr::get_sim_features()
+  sim_features_z2 <- prioritizr::get_sim_features()
+  # define feature groupings
+  x <- rep_len(c("a", "b"), terra::nlyr(sim_features))
+
+  t2 <- matrix(NA, ncol = 2, nrow = 5)
+  t2[, 1] <- .2
+  t2[, 2] <- .1
+
+  zones <- prioritizr::zones(z1 = sim_features_z1,
+                             z2 = sim_features_z2)
+
+  conf_level <- .5
+  # build problem
+  p <-
+    prioritizr::problem(c(sim_pu_raster,sim_pu_raster), zones) |>
+    add_robust_min_set_objective(method = "Chance") |>
+    prioritizr::add_relative_targets(t2) |>
+    add_constant_robust_constraints(x, conf_level) |>
+    prioritizr::add_binary_decisions() %>%    prioritizr::add_default_solver(verbose = F)
+
+  sp <- solve(p)
+
+  expect_lte(max(values(sp[[1]] + sp[[2]]), na.rm = T), 1)
 })
 
 test_that("invalid arguments", {
-  # TODO
-})
+  skip_if_not_installed("terra")
+  # import data
+  sim_pu_raster <- prioritizr::get_sim_pu_raster()
+  sim_features <- prioritizr::get_sim_features()
+  # define feature groupings
+  x <- rep_len(c("a", "b"), terra::nlyr(sim_features))
+
+  # Groups not specified
+  expect_error(
```

Comment:
here and elsewhere, you can provide a string to check that - not only is an error message returned - but an error message that contains a specific word or phrase. this helps ensure that the correct error message is returned